### PR TITLE
Fix running of auth tests

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve? 

The tests in `auth/auth_test.go` were not run.

To see the issue modify `auth/auth_test.go` to make a test case that should fail. Then run `make test`. The expected outcome is a failed test run, but in fact it doesn't show any errors.

### What is changed and how it works?

- This adds `check.TestingT()` to `auth/auth_test.go`
- This adds a simple check in `test.sh` to test for similar cases.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by dveeden** [comment]
```
[dvaneeden@dve-carbon parser]$ make test
GO111MODULE=on go build -o bin/goyacc goyacc/main.go goyacc/format_yacc.go
gofmt (simplify)
bin/goyacc -o parser.go -p yy -t Parser parser.y
Parse table entries: 2277465 of 5314635, x 16 bits == 4554930 bytes
bin/goyacc -o hintparser.go -p yyhint -t hintParser hintparser.y
Parse table entries: 16703 of 28560, x 16 bits == 33406 bytes
sh test.sh
ok  	github.com/pingcap/parser	3.498s	coverage: 75.7% of statements in ./...
ok  	github.com/pingcap/parser/ast	0.884s	coverage: 41.3% of statements in ./...
ok  	github.com/pingcap/parser/auth	0.121s	coverage: 0.8% of statements in ./... [no tests to run]
ok  	github.com/pingcap/parser/charset	0.142s	coverage: 1.0% of statements in ./...
ok  	github.com/pingcap/parser/format	0.132s	coverage: 1.3% of statements in ./...
?   	github.com/pingcap/parser/goyacc	[no test files]
ok  	github.com/pingcap/parser/model	0.142s	coverage: 2.0% of statements in ./...
```
This is the output without this PR. Note the `[no tests to run]`.

**Comment by dveeden** [comment]
Looks like #1220 has a similar fix, but doesn't change `test.sh`.

**Comment by dveeden** [comment]
Changed `test.sh` based on https://github.com/koalaman/shellcheck

File: test.sh [hunk 1]
```diff
@@ -1 +1,14 @@
+#!/bin/sh
+
+# If 'check.TestingT' is not used in any of the *_test.go files in a subdir no tests will run.
```
Comment (tangenta) [inline test.sh:3]: Thanks! Let's run this check in the `make test` of Makefile.

**Comment by tiancaiamao** [comment]
This is included in https://github.com/pingcap/parser/pull/1232 already?

**Comment by dveeden** [comment]
> This is included in #1232 already?

Both #1232 and #1236 are based on top of this commit/PR.

Note that #1220 has a similar fix but doesn't include the changes in `test.sh`

**Comment by tiancaiamao** [comment]
> > This is included in #1232 already?
> 
> Both #1232 and #1236 are based on top of this commit/PR.
> 
> Note that #1220 has a similar fix but doesn't include the changes in `test.sh`

We should have merge this one first ...
Now that #1232 is merged already, I'll close this one.

**Pull request closed without merging**